python/lsst/ts/standardscripts/auxtel/detector_characterization/get_std_flat_dataset.py

```diff
@@ -0,0 +1,279 @@
+                               f"(0.,{self.maximum_exp_time}], got {t_dark}")
+
+        if n_bias is not None and n_bias > 0:
+            self.n_bias = n_bias
+        elif n_bias is not None:
+            raise RuntimeError(f"Number of bias frames must be larger than 0, got {n_bias}")
+
+        if n_flat is not None and n_flat > 0:
+            self.n_flat = n_flat
+        elif n_flat is not None:
+            raise RuntimeError(f"Number of flat field frames must be larger than 0, got {n_flat}")
+
+        if flat_base_exptime is not None and flat_base_exptime > 0.:
+            self.flat_base_exptime = flat_base_exptime
+        elif flat_base_exptime is not None:
+            raise RuntimeError(f"Base flat field exposure time must be larger than 0., "
+                               f"got {flat_base_exptime}")
+
+        if flat_dn_range is not None:
+            self.flat_dn_range = np.array(flat_dn_range, dtype=float)
+
+        if np.any(self.flat_dn_range <= 0.):
+            n_bad = len(np.where(self.flat_dn_range <= 0.)[0])
+            raise RuntimeError(f"'flat_dn_range' must be larger than zero. "
+                               f"Got {n_bad} bad values.")
+
+        larger_flat_exptime = np.max(self.flat_dn_range)*self.flat_base_exptime
+        if larger_flat_exptime > self.maximum_exp_time:
+            raise RuntimeError(f"Flat field maximum exposure time {larger_flat_exptime}s "
+                               f"above maximum allowed {self.maximum_exp_time}s.")
+
+        if filter_id is not None:
+            self.filter = filter_id
+
+        if grating_id is not None:
+            self.grating = grating_id
+
+        if linear_stage is not None:
+            self.linear_stage = linear_stage
+
+        if read_out_time > 0.:
+            self.read_out_time = read_out_time
+        else:
+            self.log.warning(f"Read out time must be larger than 0., got {read_out_time}. "
+                             f"Using default value, {self.read_out_time}s.")
+
+    async def run(self):
+        """Run method.
+        """
+
+        await self.checkpoint("setup")
+
+        if self.filter is not None:
+            await self.checkpoint(f"Configuring ATSpec filter; {self.filter}")
+            self.atspec.cmd_changeFilter.set(filter=self.filter)
+            await self.self.atspec.cmd_changeFilter.start(timeout=self.cmd_timeout)
+
+        if self.grating is not None:
+            await self.checkpoint(f"Configuring ATSpec grating; {self.grating}")
+            self.atspec.cmd_changeDisperser.set(disperser=self.grating)
+            await self.atspec.cmd_changeDisperser.start(timeout=self.cmd_timeout)
+
+        if self.linear_stage is not None:
+            await self.checkpoint(f"Configuring ATSpec linear stage; {self.linear_stage}")
+            self.atspec.cmd_moveLinearStage.set(distanceFromHome=self.linear_stage)
+            await self.atspec.cmd_moveLinearStage.start()
```

Comment:
Please consider a timeout for this command, since waiting without a timeout may leave your script hung forever. If you are worried about really slow moves then please at least add a code comment saying so.